pysparkling/sql/expressions/mappers.py

```diff
@@ -49,14 +51,24 @@ def __init__(self, item, field):
         self.field = field
 
     def eval(self, row, schema):
-        item_value = self.item.eval(row, schema)
+        try:
+            idx = schema.names.index(self.item.col_name)
+            if isinstance(schema.fields[idx].dataType, StructType):
+                item_value = dict(list(zip(schema.fields[idx].dataType.names, self.item.eval(row, schema))))
```

Comment:
Shouldn't `self.item.eval(row, schema)` return a dict when `self.item` is a `StructType`?

Reply:
When trying to `df.select(df.r.getField('b')).collect()`, the `self.item.eval` produces a tuple.

Reply:
Yes it produces a Row, my bad, thank you